### PR TITLE
Feat/swap <img> for image component and some touch ups

### PR DESCRIPTION
**Pull request by Yoda-Soda**
# Context

[Link](https://www.notion.so/cloudcannon/Add-padding-to-nav-items-23c668d188f242a9b6da7b125c47c7e0?pvs=4) to Notion add padding to nav items
[Link](https://www.notion.so/cloudcannon/Replace-images-161b230a3b29496997df04d432fa7de8?pvs=4) to Notion swap all  for image component
[Link](https://www.notion.so/cloudcannon/Nav-animations-22c17fbc3a22478286e632b032fa8b94?pvs=4) to Notion animate mobile nav

I've add a fix where bodyscroll lock will lock up the desktop view if the mobile nav was left open and viewport changed to desktop size.

# Testing (tester)
 
Make sure to test with and with out image logos for nav and footer. 

The reviewer should check on this branch:

- The code
- The site in CloudCannon ([link](https://app.cloudcannon.com/42158/editor#sites/119070/collections/pages/:/edit?editor=visual&url=%2F&path=%2Fsrc%2Fpages%2Findex.html&collection=pages) here)
- Test site: https://new-flamingo.cloudvent.net/

# Before merge (PR owner)

- [ ] Delete the test site in CloudCannon

### CODE REVIEW

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
- The text is showing immediately and then the animation is happening on the nav dropdown
- The way the sizes is set needs some work, they don't all have the same sizes so it should be on a per component basis. Thinking we implement it but default to 100vw if the image component isn't passed a sizes param.

File: component-library/components/generic/image/image.bookshop.yml [hunk 1]
```diff
@@ -12,11 +12,13 @@ spec:
 blueprint:
   image_path: 
   image_alt:
+  image_sizes: 100vh
```
Comment (AlyshaNolan) [inline component-library/components/generic/image/image.bookshop.yml]:
```suggestion
  image_sizes: 
```

File: component-library/components/generic/image/image.bookshop.yml [hunk 2]
```diff
@@ -12,11 +12,13 @@ spec:
...
 
 # Overrides any fields in the blueprint when viewing this component in the component browser
 preview:
   image_path: /assets/images/Testimonial--background-feature.png
   image_alt: A woman laughing
+  image_sizes: 100vh
```
Comment (AlyshaNolan) [inline component-library/components/generic/image/image.bookshop.yml]:
```suggestion
  image_sizes: 100vw
```

File: component-library/components/generic/image/image.bookshop.yml (hunk 1 above)
Comment (AlyshaNolan) [reply component-library/components/generic/image/image.bookshop.yml]: Otherwise it will always have 100vh

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
- [x] The text is showing immediately and then the animation is happening on the nav dropdown
- [x] Image sizes should be a hidden input

**Comment by Yoda-Soda** [comment]
I've added a fade in to the text

**Review by aleksandrgordeev — DISMISSED** [review]
The bodyscroll lock in mobile view allows to scroll down which makes menu button unavailable. I've attached a video to demonstrate:

https://github.com/CloudCannon/small-business-template-eleventy/assets/69171343/0d9d2f3c-c39d-4176-85c0-2b596439c1e0

**Comment by Yoda-Soda** [comment]
> The bodyscroll lock in mobile view allows to scroll down which makes menu button unavailable. I've attached a video to demonstrate:
> 
>  Screen.Recording.2023-11-24.at.16.49.38.mov

That's very odd as I can't reproduce this

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
- The fade out on the nav items isn't working
- The fade is also happening on desktop nav

I also cannot reproduce the problem that Aleks was having

**Comment by Yoda-Soda** [comment]
> The fade out on the nav items isn't working
> The fade is also happening on desktop nav
> I also cannot reproduce the problem that Aleks was having

Add the fade for the text. 

Couldn't recreate the fade happening on the desktop. But it's working now so yeah 🥳

**Review by AlyshaNolan — APPROVED** [review]

**Comment by AlyshaNolan** [comment]
Approved with the condition of blocks being removed from the index page, and the icon put back to original (and the uploaded one being deleted)

**Pull request merged**